packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java

```diff
@@ -1323,6 +1323,25 @@ public void scrollTo(int x, int y) {
     setPendingContentOffsets(x, y);
   }
 
+  /**
+   * Scrolls to a new position preserving any momentum scrolling animation.
+   */
+  public void scrollToPreservingMomentum(int x, int y) {
```

Comment:
I think we have this logic now copied into a few places. E.g. we need to do something like this for RTL adjustment: https://github.com/facebook/react-native/blob/d74ecb3adced7f5b26895d6c770cd3e7f2d7aa66/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java#L1419

I don't remember why I needed to do some of the jankier bits there.